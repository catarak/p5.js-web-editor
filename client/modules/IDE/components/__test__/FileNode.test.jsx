import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import { FileNode } from '../FileNode';


describe('FileNode', () => {
  const props = {
    id: '123456',
    parentId: 'abcdef',
    children: [],
    name: 'test.js',
    fileType: 'file',
    isSelectedFile: false,
    isFolderClosed: false,
    setSelectedFile: jest.fn(),
    deleteFile: jest.fn(),
    updateFileName: jest.fn(),
    resetSelectedFile: jest.fn(),
    newFile: jest.fn(),
    newFolder: jest.fn(),
    showFolderChildren: jest.fn(),
    hideFolderChildren: jest.fn(),
    canEdit: false
  };

  it('it renders file node', () => {
    const getWrapper = () => shallow(<FileNode {...props} />);
    const filenode = getWrapper();
    expect(filenode.exists('.file-item__content')).toEqual(true);
  });

  it('check selected file node displayed', () => {
    const getWrapper = () => shallow(<FileNode {...props} isSelectedFile />);
    const filenode = getWrapper();
    expect(filenode.exists('.sidebar__file-item--selected')).toEqual(true);
  });

  it('check selection of file on click', () => {
    const getWrapper = () => shallow(<FileNode {...props} isSelectedFile />);
    const fileNode = getWrapper();
    const nameButton = fileNode.find('.sidebar__file-item-name');
    nameButton.simulate('click', { stopPropagation() {} });
    expect(fileNode.exists('.sidebar__file-item--selected')).toEqual(true);
  });

  it('renders correctly', () => {
    const tree = renderer
      .create(<FileNode {...props} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
