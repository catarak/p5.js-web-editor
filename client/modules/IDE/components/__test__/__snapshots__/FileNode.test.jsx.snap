// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FileNode renders correctly 1`] = `
<div
  className="sidebar__file-item"
>
  <div
    className="file-item__content"
    onContextMenu={[Function]}
  >
    <span
      className="file-item__spacer"
    />
    <span
      className="sidebar__file-item-icon"
    >
      <span
        className="isvg loading"
      />
    </span>
    <button
      className="sidebar__file-item-name"
      onClick={[Function]}
    >
      test.js
    </button>
    <input
      className="sidebar__file-item-input"
      onBlur={[Function]}
      onChange={[Function]}
      onKeyPress={[Function]}
      type="text"
      value="test.js"
    />
    <button
      aria-label="view file options"
      className="sidebar__file-item-show-options"
      onBlur={[Function]}
      onClick={[Function]}
      tabIndex="0"
    >
      <span
        className="isvg loading"
      />
    </button>
    <div
      className="sidebar__file-item-options"
    >
      <ul
        title="file options"
      >
        <li>
          <button
            className="sidebar__file-item-option"
            onClick={[Function]}
          >
            Rename
          </button>
        </li>
        <li>
          <button
            className="sidebar__file-item-option"
            onClick={[Function]}
          >
            Delete
          </button>
        </li>
      </ul>
    </div>
  </div>
  <ul
    className="file-item__children"
  />
</div>
`;
